### PR TITLE
[WIP] Allow empty enumeration option when non-required

### LINKED ISSUE #2471: Enumeration value is not set by default
**Informations**
- **Node.js version**: v10.0.0
- **NPM version**: v6.0.0
- **Strapi version**: v3.0.0-alpha.16
- **Database**: MongoDB
- **Operating system**: macOS

**What is the current behavior?**
**Steps to reproduce the problem**

I have `Battle` model with an `stage` attribute like this, because is an optional field:

```
"stage": {
  "default": "",
  "type": "enumeration",
  "enum": [
    "",
    "final",
    "semiFinal",
    "thirdAndFourthPlace",
    "quarterFinals",
    "roundOf16",
    "groupStage"
  ]
},
```

It works with this query:

```
{
  battles {
    _id
    team1 {
      slug: username
    }
    team2 {
      slug: username
    }
  }
}
```

But it doesn't work if I add the `stage` attribute:

```
{
  battles {
    _id
    team1 {
      slug: username
    }
    team2 {
      slug: username
    }
    stage
  }
}
```

The error is:

> `"message": "Expected a value of type \"ENUM_BATTLE_STAGE\" but received: \"\"",`

**What is the expected behavior?**

Allow empty `stage`, because it is not required and the default value is and empty string (`""`).

### PR DESCRIPTION
**Pull request by abdonrd**
My PR is a:
- [ ] 💥 Breaking change
- [x] 🐛 Bug fix #2471
- [ ] 💅 Enhancement
- [ ] 🚀 New feature

Main update on the:
- [x] Admin
- [ ] Documentation
- [ ] Framework
- [ ] Plugin

### CODE REVIEW

File: packages/strapi-plugin-content-manager/admin/src/components/Edit/index.js [hunk 1]
```diff
@@ -55,6 +55,40 @@ const getInputType = (type = '') => {
...
+          const details = get(this.props.schema, ['editDisplay', 'availableFields', attr]);
+
+          const layout = this.getInputLayout(attr);
+          const appearance = get(layout, 'appearance');
+          const type = !isEmpty(appearance)
+            ? appearance.toLowerCase()
+            : get(layout, 'type', getInputType(details.type));
+
+          if (type === 'select') {
+            const attribute = get(this.props.schema, ['attributes', attr]);
```
Comment (soupette) [inline packages/strapi-plugin-content-manager/admin/src/components/Edit/index.js]: Here's my work so far:

```
 const value = attribute.default ? attribute.default : attribute[0];
            // Apparently you can't set the value to either null or '' if this options isn't set in your model

            const target = {
             type: 'select',
             name: attr,
             value,
           };
```

Comment (abdonrd) [reply packages/strapi-plugin-content-manager/admin/src/components/Edit/index.js]: Mmm but that's the reason why we try to programmatically add an empty option if the field is not required, right?

From:
screenshot 2018-12-24 at 10 03 47

To:
screenshot 2018-12-24 at 10 05 13

Comment (soupette) [reply packages/strapi-plugin-content-manager/admin/src/components/Edit/index.js]: Well with the new code we programmatically set the value if the user didn't set a default one in his model.

Comment (abdonrd) [reply packages/strapi-plugin-content-manager/admin/src/components/Edit/index.js]: Updated wit your code!

**Comment by soupette** [comment]
@abdonrd Are you still working on this PR or is the feature ready? We are locking all the upcoming PR so I think that I could re implement this feature when we extend the plugin for the repeatable fields.

**Pull request closed without merging**